packages/runtime/src/types.ts

```diff
@@ -235,14 +237,54 @@ export enum TransactionType {
   CallNoOpSSC,
   ClearSSC,
   CloseSSC,
-  DeleteSSC
+  DeleteSSC,
+  DeployASA,
+  DeploySSC,
+  OptInASA,
+  OptInSSC
 }
 
 export interface Sign {
   sign: SignType
   lsig?: LogicSig
 }
 
+export interface DeployASAParam extends Sign {
+  type: TransactionType.DeployASA
+  fromAccount: AccountSDK
+  asaName: string
+  asaDef?: ASADef
```

Comment:
This should be `Partial<ASADef>`. User can pass only few params (which he wants to override) in execParams

Reply:
same